apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultInjector.java

```diff
@@ -60,7 +63,12 @@ protected void configure() {
       bind(ConfigRegistry.class).to(DefaultConfigRegistry.class).in(Singleton.class);
       bind(ConfigFactory.class).to(DefaultConfigFactory.class).in(Singleton.class);
       bind(ConfigUtil.class).in(Singleton.class);
-      bind(HttpUtil.class).in(Singleton.class);
+      bind(HttpUtil.class).toProvider(new Provider<HttpUtil>() {
+        @Override
+        public HttpUtil get() {
+          return ServiceBootstrap.loadPrimary(HttpClientFactory.class).createHttpClient();
```

Comment:
I'm thinking how about simply `ServiceBootstrap.loadPrimary(HttpUtil.class)` and make `HttpUtil` implement `Ordered`? Then this can be a common solution to make other interfaces easily to be customized in the future, such as `ConfigManager`?